pkg/autoscaler/fake/fakes.go

```diff
@@ -30,3 +34,23 @@ func (mtp *ManualTickProvider) NewTicker(time.Duration) *time.Ticker {
 		C: mtp.Channel,
 	}
 }
+
+// C returns the tick channel.
+func (mtp *ManualTickProvider) C() <-chan time.Time {
+	return mtp.Channel
+}
+
+// Stop is nonce here.
+func (mtp *ManualTickProvider) Stop() {}
+
+// Clock is K8s clock.Clock but it overrides tick provider
+// with ManualTickProvider above.
+type Clock struct {
+	*clock.FakeClock
+	TP *ManualTickProvider
+}
+
+// NewTicker returns a NewTicker which is a ManualTickProvider.
+func (fc Clock) NewTicker(time.Duration) clock.Ticker {
```

Comment:
As `TP` is exported var, could we just use it directly without this func?

Reply:
No we need to adhere to a specific interface.